### PR TITLE
[ONNX] Update slice symbolic function

### PR DESCRIPTION
**Pull request by KsenijaS**
During scripting, combination of shape (or size()) and slice (e.g x.shape[2:]) produces following error:
 slice() missing 1 required positional argument: 'step' 
This happens because aten::slice has 2 signatures:

- aten::slice(Tensor self, int dim, int start, int end, int step) -> Tensor
- aten::slice(t[] l, int start, int end, int step) -> t[]

and when a list is passed instead of tensor the 2nd of the two slice signatures is called, and since it has 4 instead of 5 arguments it produces the above exception.

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -902,11 +902,21 @@ def test_slice_dynamic_script(self):
         class DynamicSliceModel(torch.jit.ScriptModule):
             @torch.jit.script_method
             def forward(self, x):
-                return x[1:x.size(0)]
+                return x[1:x.size(1)]
 
         x = torch.rand(1, 2)
         self.run_test(DynamicSliceModel(), x)
 
+    @skipIfUnsupportedMinOpsetVersion(10)
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:910]: I know you have enables this for scripting. But if this test is already passing tracing as well, could you please write it as a nn.Module and use enableScriptTest decorator?
I think going forward we can use this approach for all tests. We want to make sure both code-paths are supported, unless there's a known issue.

File: torch/onnx/symbolic_opset10.py [hunk 2]
```diff
@@ -137,8 +137,11 @@ def __interpolate(g, input, size, scale_factor, mode , align_corners, recompute_
 
 def _slice(g, input, axes, starts, ends, steps=None, dynamic_slice=False):
     if dynamic_slice:
+        print("start, end", starts, ends)
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset10.py:140]: Remove debug code.

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -1597,26 +1597,38 @@ def eye(g, n, m, dtype=None, layout=None, device=None, pin_memory=False):
...
-            raise RuntimeError('Unsupported: ONNX export of Slice with dynamic inputs. DynamicSlice '
-                               'is a deprecated experimental op. Please use statically allocated '
-                               'variables or export to a higher opset version.')
+
+def slice(g, self, *args):
+    if len(args) == 4:
+        # aten::slice(Tensor self, int dim, int start, int end, int step) -> Tensor
+        dim, start, end, step = args
+        if step != 1:
+            _unimplemented("slice", "step!=1 is currently not supported")
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset9.py:1606]: Please keep consistent error handling. 
Maybe use RuntimeError like line 1610

File: torch/onnx/symbolic_opset10.py [hunk 4]
```diff
@@ -137,8 +137,11 @@ def __interpolate(g, input, size, scale_factor, mode , align_corners, recompute_
...
         starts = g.op("Unsqueeze", starts, axes_i=[0])
         ends = g.op("Unsqueeze", ends, axes_i=[0])
+        if isinstance(axes, int):
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset10.py:143]: What if dynamic_slice = False and axes is an integer?
Would it be a case we need to check?

**Review by neginraoof — COMMENTED** [review]
Left some comments. Thanks.

File: torch/onnx/symbolic_opset10.py (hunk 4 above)
Comment (KsenijaS) [reply torch/onnx/symbolic_opset10.py:143]: when dynamic_slice is False axes should be an integer. This check is only needed when dynamic_slice is True, because Unsqueeze operation cannot be performed on an int.

**Comment by neginraoof** [comment]
Thanks. Can you add more info about the new slice signature you're adding as part of the PR description?

**Comment by neginraoof** [comment]
@KsenijaS test_slice is failing. Please take a look.

**Review by neginraoof — APPROVED** [review]
Looks good.

**Comment by KsenijaS** [comment]
@bzinodev can we please merge this? Thanks!

File: torch/onnx/symbolic_opset9.py [hunk 5]
```diff
@@ -1613,26 +1613,39 @@ def eye(g, n, m, dtype=None, layout=None, device=None, pin_memory=False):
...
-            end_unsqueezed = g.op("Unsqueeze", end, axes_i=[0])
-            dim_unsqueezed = g.op("Unsqueeze", dim, axes_i=[0])
-            return g.op("DynamicSlice", self, start_unsqueezed, end_unsqueezed, dim_unsqueezed)
-    else:
+            start = _parse_arg(start, 'i')
+            end = _parse_arg(end, 'i')
+            dim = _parse_arg(dim, 'i')
+            return sym_help._slice_helper(g, self, axes=[dim], starts=[start], ends=[end])
+    elif len(args) == 3:
+        # aten::slice(t[] l, int start, int end, int step) -> t[]
```
Comment (houseroad) [inline torch/onnx/symbolic_opset9.py:1641]: interesting, I am curious where this aten function is defined?

Comment (KsenijaS) [reply torch/onnx/symbolic_opset9.py:1641]: it's defined in torch/csrc/jit/runtime/register_prim_ops.cpp: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/register_prim_ops.cpp#L503
and it's calling listSlice function defined here: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/register_ops_utils.cpp#L429

**Review by bzinodev — APPROVED** [review]

**Comment by KsenijaS** [comment]
@bzinodev I see there are some internal Facebook tests failing? Can you share what the failures are? Can we merge athis PR despite those failures?
